### ISSUE TITLE
GET request params in data are ignored

### ISSUE THREAD

**Issue by mallorymallory** [veox/python3-krakenex#140]
### Versions

```
OS:  any     
Python:   any
krakenex: 2.2.0
```

### What are you trying to achieve?

``` python
import krakenex
k=krakenex.API()
k.query_public("Ticker", {"pair":"XBTUSD"}

```

### What do you expect to happen?
market data for requested pair returned

### What happens instead?
market data for all pairs returned
params for GET being sent in session.get data are ignored by Kraken - need to be urlecoded

k.query_public("Ticker?pair=XBTUSD"} 
works^ 

``` python
# error message

```

**Comment by jj-elliott-animus** [comment]
Yea, I noticed this as well. It seems that Kraken isn't respecting params sent in the GET body. 
I've changed my local to:
`# Since January 2024, public endpoints only support GET.
        if "/public/" in urlpath:
            self.response = self.session.get(
                url, params=data, headers=headers, timeout=timeout
            )
        else:
            self.response = self.session.post(
                url, data=data, headers=headers, timeout=timeout
            )

        if self.response.status_code not in (200, 201, 202):
            self.response.raise_for_status()`
            
as a temporary fix and it seems to be working.

**Comment by egner** [comment]
Same problem here. For now, I go with this workaround in my client code:
```
data_str = '&'.join([f'{k}={v}' for k, v in data.items()])
method_data = f'{method}?{data_str}' if len(data_str) > 0 else method
response = api.query_public(method_data, timeout=timeout)
 ```

**Comment by veox** [comment]
@ jj-elliott-animus Very clean workaround!

I'll go review the PR #141 now.

**Comment by veox** [comment]
Closing as fixed; bugfix release 2.2.1 on the way.

**Comment by veox** [comment]
Closing as fixed; bugfix release 2.2.1 on the way.

**Comment by veox** [comment]
Post-mortem: this wasn't caught because CI hasn't been working for a while now. :/

The repo has been degrading in "maintenance mode" and I haven't been checking github regularly, for years now. Lucky I did yesterday...
